packages/store/src/lib/batch.ts

```diff
@@ -11,10 +13,25 @@ export function emitOnce<T>(cb: () => T) {
   if (!batchInProgress.getValue()) {
     batchInProgress.next(true);
     const value = cb();
-    batchInProgress.next(false);
+    if (asyncBatchesInProgress === 0) {
+      batchInProgress.next(false);
+    }
 
     return value;
   }
 
   return cb();
 }
+
+export async function emitOnceAsync<T>(cb: () => Promise<T>) {
```

Comment:
I would support both observable and promise

Reply:
Sorry for the late response, I was not home for a few days. I added the possibility of passing an observable.

I decided to update the store when the observable emits its first value, not when it completes. Otherwise, I assume it would lead to unexpected behavior. Do you agree with that?